internal/form/submit/service_test.go

```diff
@@ -0,0 +1,257 @@
+				qm := mocks.NewMockQuestionStore(t)
+				rm := mocks.NewMockFormResponseStore(t)
+
+				qm.EXPECT().
+					ListByFormID(mock.Anything, p.formID).
+					Return(p.qs, nil).Once()
+
+				p.qStore = qm
+				p.rStore = rm
+				p.service = submit.NewService(zap.NewNop(), qm, rm)
+				return context.Background()
+			},
+			validate: func(t *testing.T, p Params, got response.FormResponse, errs []error) {
+				require.NotNil(t, errs)
+				require.Len(t, errs, 1)
+				require.Equal(t, uuid.Nil, got.ID)
+				p.rStore.AssertNotCalled(t, "CreateOrUpdate", mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything)
+			},
+		},
+		{
+			name: "Return error and do not call CreateOrUpdate if ListByFormID fails",
+			params: Params{
+				formID:  formID,
+				userID:  userID,
+				answers: []shared.AnswerParam{ans(q1, "x")},
+			},
+			expectedErr: true,
+			setup: func(t *testing.T, p *Params) context.Context {
+				qm := mocks.NewMockQuestionStore(t)
+				rm := mocks.NewMockFormResponseStore(t)
+
+				qm.EXPECT().
+					ListByFormID(mock.Anything, p.formID).
+					Return(nil, errors.New("db down")).Once()
+
+				p.qStore = qm
+				p.rStore = rm
+				p.service = submit.NewService(zap.NewNop(), qm, rm)
+				return context.Background()
+			},
+			validate: func(t *testing.T, p Params, got response.FormResponse, errs []error) {
+				require.NotNil(t, errs)
+				require.Equal(t, uuid.Nil, got.ID)
+				p.rStore.AssertNotCalled(t, "CreateOrUpdate", mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything)
+			},
+		},
+		{
+			name: "Return error if validation passes but CreateOrUpdate fails",
+			params: Params{
+				formID:  formID,
+				userID:  userID,
+				qs:      []question.Answerable{makeQ(q1, question.QuestionTypeShortText, nil)},
+				answers: []shared.AnswerParam{ans(q1, "ok")},
+			},
+			expectedErr: true,
+			setup: func(t *testing.T, p *Params) context.Context {
+				qm := mocks.NewMockQuestionStore(t)
+				rm := mocks.NewMockFormResponseStore(t)
+
+				qm.EXPECT().
+					ListByFormID(mock.Anything, p.formID).
+					Return(p.qs, nil).Once()
+
+				rm.EXPECT().
+					CreateOrUpdate(
+						mock.Anything,
+						p.formID,
+						p.userID,
+						p.answers,
+						[]response.QuestionType{response.QuestionType(question.QuestionTypeShortText)},
+					).
+					Return(response.FormResponse{}, errors.New("insert fail")).
+					Once()
+
+				p.qStore = qm
+				p.rStore = rm
+				p.service = submit.NewService(zap.NewNop(), qm, rm)
+				return context.Background()
+			},
+			validate: func(t *testing.T, p Params, got response.FormResponse, errs []error) {
+				require.NotNil(t, errs)
+				require.Equal(t, uuid.Nil, got.ID)
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ctx := context.Background()
+			params := tc.params
+			if tc.setup != nil {
+				ctx = tc.setup(t, &params)
+			}
+
+			got, errs := params.svc.Submit(ctx, params.formID, params.userID, params.answers)
```

Comment:
The service name should be updated
```suggestion
			got, errs := params.service.Submit(ctx, params.formID, params.userID, params.answers)
```